StringPrototypeExtension.js

```diff
@@ -0,0 +1,185 @@
+String.prototype.hasVowels = function() {
+  var vowelRegex = /[aeiou]/gi;
+  var hasVowel = vowelRegex.test(this);
+  return hasVowel;
+};
+
+String.prototype.isQuestion = function() {
+  var isQuestionRegex = /\w+[?]$/;
+  return isQuestionRegex.test(this);
+};
+
+String.prototype.numberWords = function() {
+  var numberWordRegex = [
+    "zero",
+    "one",
+    "two",
+    "three",
+    "four",
+    "five",
+    "six",
+    "seven",
+    "eight",
+    "nine"
+  ];
+  var digitsArray = this.match(/\d/g);
+  var numberToWord = "";
+  digitsArray.forEach(digit => {
+    numberToWord += numberWordRegex[digit].concat(" ");
+  });
+  return numberToWord;
+};
+
+String.prototype.wordsToArray = function() {
+  var wordRegex = /\b[0-9a-zA-Z_\-?]+\b/g;
+  var wordArray = this.match(wordRegex);
+
+  return wordArray;
+};
+
+String.prototype.wordCount = function() {
+  return this.wordsToArray().length;
+};
+
+String.prototype.isDigit = function() {
+  var digitRegex = /\d/g;
+
+  if (!digitRegex.test(this)) {
+    return false;
+  }
+  return this.match(digitRegex).length === 1 ? true : false;
+};
+
+String.prototype.toCurrency = function() {
+  var regexPattern = /^\d+(\.\d+)?$/g;
+  if (!regexPattern.test(this)) {
+    return false;
+  }
+  var currency = Number(this)
+    .toFixed(2)
+    .replace(/\d(?=(\d{3})+\.)/g, "$&,");
```

Comment:
I think it is better if the regex pattern passed as the first parameter to the replace function is extracted into a well-named variable. This way, the replace function is given more context as to what it does.